napari/_qt/containers/qt_layer_model.py

```diff
@@ -51,7 +51,7 @@ def setData(
         role: int = Qt.ItemDataRole.EditRole,
     ) -> bool:
         if role == Qt.ItemDataRole.CheckStateRole:
-            self.getItem(index).visible = value
+            self.getItem(index).visible = value == Qt.Checked
```

Comment:
When reproducing the bug in https://github.com/napari/napari/issues/1346, I saw that `Layer.editable == 2`, which is referring the Qt checked state enum nominal value.

Reply:
Wow! LOL In that case I might suggest:

```suggestion
            # Qt.Checked is a value of the Qt.CheckedState enum, and its
            # nominal value is 2. We want to set this to a boolean, so we use
            # value == Qt.Checked.
            # See: https://doc.qt.io/qt-5/qt.html#CheckState-enum
            self.getItem(index).visible = value == Qt.Checked
```

Reply:
And here we Why code coverage is important, especially in the context of differences between PyQt5/PySide2/PyQt6/PySide6 

This code will crash at least on PySide6 and maybe on PyQt6 as it refers to enum value without using enum value. 

```suggestion
            self.getItem(index).visible = value == Qt.CheckState.Checked
```

Reply:
Ouch, tricky.

Reply:
Thanks for the catch. Reproduced on pyside6. Also added some tests to cover the relationship between the checked state of items in the Qt layer list and `Layer.visible`.

Both suggestions are useful, so I just combined myself in a commit rather than deal with github-suggestion-merge conflicts.